### ISSUE TITLE
Security_Log_Profile > network > storageFormat value returns "no such variable"

### ISSUE THREAD

**Issue by mofftech** [F5Networks/f5-appsvcs-extension#159]
##### Do you already have an issue opened with F5 support?

SR 1-5500401169.

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

3.14.0

##### BIGIP VERSION

14.1.2

##### SUMMARY

We're attempting a workaround to add a user defined security log profile format.

We want to define user-defined format of security log profile using AS3. The following is the format in tmsh.

```
security log profile Shared/pf_log_01 {
    network {
        undefined {
            filter {
                log-acl-match-accept enabled
                log-acl-match-drop enabled
                log-translation-fields enabled
            }
            format {
                type user-defined
                user-defined "${date_time},${bigip_hostname},${management_ip_address},${src_ip},${src_port},${dest_ip},${dest_port},${translated_src_ip},${translated_dest_ip},${translated_src_port},${translated_dest_port},${date_time},,${protocol},${action}"
            }
            publisher Shared/lp_hsl_01
        }
    }
}
```

C3001273 was raised and ID791469 was filed.

As a workaround, the following configuration was attempted:

```
{
    "class": "ADC",
    "schemaVersion": "3.13.0",
    "id": "Security_Log_Profile",
    "TEST_Security_Log_Profile": {
        "class": "Tenant",
        "Shared": {
            "class": "Application",
            "template": "shared",
            "pf_log_01": {
                "class": "Security_Log_Profile",
                "network": {
                    "storageFormat": "${date_time},${bigip_hostname}"
                }
            }
        }
    }
}
```

It seems there is a different issue with "network". Now we get the error below:

> "response": "can't read \\\"bigip_hostname\\\": no such variable",

I was requested by F5 PD to raise an issue via github.

##### STEPS TO REPRODUCE

```
{
    "class": "ADC",
    "schemaVersion": "3.13.0",
    "id": "Security_Log_Profile",
    "TEST_Security_Log_Profile": {
        "class": "Tenant",
        "Shared": {
            "class": "Application",
            "template": "shared",
            "pf_log_01": {
                "class": "Security_Log_Profile",
                "network": {
                    "storageFormat": "${date_time},${bigip_hostname}"
                }
            }
        }
    }
}
```

##### EXPECTED RESULTS

Bespoke storageFormat options added to network security log profile.

##### ACTUAL RESULTS

"response": "can't read \\\"bigip_hostname\\\": no such variable",

**Comment by focrensh** [comment]
AT-808 created internally to track this

**Comment by ianwijaya** [comment]
Hi, 

I encountered the same issue. the error is gone after I added "\" character before "$"
`"storageFormat": "action=\${action}"`
however, I got different error. 

```
TASK [roles/appconf : URI POST Tenant] *****************************************
fatal: [localhost]: FAILED! => {"cache_control": "no-store, no-cache, must-revalidate", "changed": false, "connection": "close", "content": "{\"results\":[{\"code\":422,\"message\":\"declaration failed\",\"response\":\"\\\\\\\"format.user-defined\\\\\\\" may not be specified with \\\\\\\"format.field-list-delimiter\\\\\\\"\",\"host\":\"localhost\"
```
Can you help to test this on your side?  I couldn't find a way to omit field-list-delimiter.

**Comment by dstokesf5** [comment]
This issue was resolved in the 3.16.0 release.
